comm/src/main/scala/coop/rchain/comm/rp/ProtocolHelper.scala

```diff
@@ -25,11 +25,7 @@ object ProtocolHelper {
       .withUdpPort(n.endpoint.udpPort)
       .withTcpPort(n.endpoint.tcpPort)
 
-  def sender(proto: Protocol): Option[PeerNode] =
-    for {
-      h <- proto.header
-      s <- h.sender
-    } yield toPeerNode(s)
+  def sender(proto: Protocol): Option[PeerNode] = Some(toPeerNode(proto.header.sender))
```

Comment:
This function just wraps result in `Some` value. This can be always done by the caller, so it should just return the value.